### PR TITLE
Add more cache benchmarks and run them in CI

### PR DESCRIPTION
**Pull request by martinbonnin**
This PR adds Cache benchmarks based on @sebj data. 

Thanks a lot to him for providing real life data to monitor the cache performance 💙 

It also adds a new CI target that runs the benchmarks in Firebase Test Lab.

### CODE REVIEW

File: scripts/run-benchmarks.main.kts [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+import okio.buffer
+import okio.source
+import java.io.File
+import java.util.Date
+
+
+/**
+ * This script expects:
+ *
+ * - `gcloud` in the path
```
Comment (BoD) [inline scripts/run-benchmarks.main.kts:28]: Is gcloud present by default on the GH actions image, or did I miss something?

**Review by BoD — APPROVED** [review]

File: scripts/run-benchmarks.main.kts (hunk 1 above)
Comment (martinbonnin) [reply scripts/run-benchmarks.main.kts:28]: Good question. Looks like [it is preinstalled indeed](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md#cli-tools)

Comment (BoD) [reply scripts/run-benchmarks.main.kts:28]: Handy! 👍

**Comment by martinbonnin** [comment]
On hold for https://github.com/gradle/gradle/issues/21125

**Comment by martinbonnin** [comment]
> On hold for https://github.com/gradle/gradle/issues/21125

Never mind, we can use AGP alpha04 for now so we can stick with Gradle 7.4.2. Marking as ready for review again

**Review by BoD — APPROVED** [review]

**Pull request merged**